compass_app/app/lib/domain/models/booking/booking_summary.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
```

Comment:
Question: What's the difference/use case for Booking, BookingSummary, and ItineraryConfig? I'm not criticizing, just curious and having trouble making sense of it on my own.

Reply:
Hi! I hope this helps:

- `Booking` (app model): Represents a full booking, including full `Destination` and `Activity` data. Used by the `/booking` screen to display the booking (pictures, tags, etc.). Created from the `Booking` API model class + `Destination` + `Activity` data classes.
- `Booking` (API model): Same concept, but the `Destination` is missing, instead, the destination id (reference) is present. Same for the `Activity` data, only the activity id (reference) is present in the object. The API model also has a `name` property, to facilitate display in the home screen.
- `BookingSummary`: Represents a booking list entry in the home screen. Only contains the necessary data to display an item. This is created from the `Booking` API model data class.
- `ItineraryConfig`: Represents the user preferences while creating a new booking, this data model can be incomplete while the user is doing changes, and it is only found locally, not on the server. Once the user finishes this process, the stored `ItineraryConfig` is used to create a full `Booking` object.

Reply:
Okay, this makes sense to me. thank you!